main.cpp

```diff
@@ -44,65 +44,105 @@ char* screenName = (char*)":0.0";
 int irqPin = 6;
 char* calibFileName = (char*)"calibpoints.cal";
 int waitTime = 5;
+int delayTime = 2000;
 
 bool calibrationMode = false;
 
-void penInterrupt(void) 
+point getPosition(void)
 {
-	usleep(10000); //bruteforce debounce
-	
+	if (digitalRead(irqPin))
+	{
+		return point(-1,-1);
+	}
+
 	point resistance = ads->getXYdata();
-	
-	if(!digitalRead(irqPin) && resistance != point(0x000, 0xFFF))
+	if (resistance == point(0x000, 0xFFF))
 	{
-		//Pen down		
-		point position = touchCalib.getDisplayPoint(resistance);
-		
-		XTestFakeMotionEvent (display, 0, position.x(), position.y(), CurrentTime );
-		XFlush(display);
-		
-		XTestFakeButtonEvent(display, 1, True, CurrentTime);
-		XFlush(display);
-		
-		while(!digitalRead(irqPin) && resistance != point(0x000, 0xFFF))
+		return point(-1,-1);
+	}
+
+	return touchCalib.getDisplayPoint(resistance);
+}
+
+point getSmoothPosition(void)
+{
+	int validCount = 0, xAvg = 0, yAvg = 0;
+	for(int i = 0; i < 9; i++) // Odd number to prevent tie
+	{
+		point position = getPosition();
+
+		if (position.x() >= 0)

```

Comment:
Indeed, as no non-linear stuff happens in the transformation, averaging one or the other is just a matter of accuracy, depending of whether you scale up or down.

My boss wants capacitive rather than resistive touch, so I won't spend any time on this anymore. You can use it however you want.